No bean could be found in the registry for: connector-rest-swagger-http4 of type: org.apache.camel.spi.RestProducerFactory

## This is a...

<pre><code>
[ ] Feature request
[x] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>

## Description
Error in the pod log during the deploying integration AMQ -> REST API (todo app) which is described in the [Fuse online tutorials](https://access.redhat.com/documentation/en-us/red_hat_fuse/7.0/html-single/ignite_sample_integration_tutorials/#amq-to-rest-api).
It has never be published and it is stuck in the 'starting' state.

I have tried it on the _syndesis-1.5.6.fuse-720003_ tag and there it works.

Log:
```
Error starting ApplicationContext. To display the auto-configuration report re-run your application with 'debug' enabled.
2018-11-12 11:38:05.702 ERROR 1 --- [           main] o.s.boot.SpringApplication               : Application startup failed
org.apache.camel.RuntimeCamelException: org.apache.camel.FailedToCreateRouteException: Failed to create route -LR6gW5c-5Suhf6Okebp: Route(-LR6gW5c-5Suhf6Okebp)[[From[sjms-0-0]] -> [SetHeader[S... because of Failed to create Producer for endpoint: swagger-operation-0-3?operationId=operation-1. Reason: org.apache.camel.NoSuchBeanException: No bean could be found in the registry for: connector-rest-swagger-http4 of type: org.apache.camel.spi.RestProducerFactory
	at org.apache.camel.util.ObjectHelper.wrapRuntimeCamelException(ObjectHelper.java:1830) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.spring.SpringCamelContext.start(SpringCamelContext.java:136) ~[camel-spring-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.spring.SpringCamelContext.onApplicationEvent(SpringCamelContext.java:174) ~[camel-spring-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:393) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:347) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:883) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:144) ~[spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:546) ~[spring-context-4.3.19.RELEASE.jar!/:4.3.19.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122) ~[spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1118) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1107) [spring-boot-1.5.16.RELEASE.jar!/:1.5.16.RELEASE]
	at io.syndesis.example.Application.main(Application.java:13) [classes!/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_151]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_151]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_151]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_151]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:50) [project-0.1-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.PropertiesLauncher.main(PropertiesLauncher.java:595) [project-0.1-SNAPSHOT.jar:na]
Caused by: org.apache.camel.FailedToCreateRouteException: Failed to create route -LR6gW5c-5Suhf6Okebp: Route(-LR6gW5c-5Suhf6Okebp)[[From[sjms-0-0]] -> [SetHeader[S... because of Failed to create Producer for endpoint: swagger-operation-0-3?operationId=operation-1. Reason: org.apache.camel.NoSuchBeanException: No bean could be found in the registry for: connector-rest-swagger-http4 of type: org.apache.camel.spi.RestProducerFactory
	at org.apache.camel.impl.RouteService.warmUp(RouteService.java:147) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.DefaultCamelContext.doWarmUpRoutes(DefaultCamelContext.java:3947) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.DefaultCamelContext.safelyStartRouteServices(DefaultCamelContext.java:3854) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.DefaultCamelContext.doStartOrResumeRoutes(DefaultCamelContext.java:3640) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.DefaultCamelContext.doStartCamel(DefaultCamelContext.java:3492) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.DefaultCamelContext.access$000(DefaultCamelContext.java:209) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.DefaultCamelContext$2.call(DefaultCamelContext.java:3251) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.DefaultCamelContext$2.call(DefaultCamelContext.java:3247) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.DefaultCamelContext.doWithDefinedClassLoader(DefaultCamelContext.java:3270) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.DefaultCamelContext.doStart(DefaultCamelContext.java:3247) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:3163) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.spring.SpringCamelContext.start(SpringCamelContext.java:133) ~[camel-spring-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	... 24 common frames omitted
Caused by: org.apache.camel.FailedToCreateProducerException: Failed to create Producer for endpoint: swagger-operation-0-3?operationId=operation-1. Reason: org.apache.camel.NoSuchBeanException: No bean could be found in the registry for: connector-rest-swagger-http4 of type: org.apache.camel.spi.RestProducerFactory
	at org.apache.camel.impl.ProducerCache.doGetProducer(ProducerCache.java:584) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.ProducerCache.acquireProducer(ProducerCache.java:168) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.processor.SendProcessor.doStart(SendProcessor.java:248) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.processor.WrapProcessor.doStart(WrapProcessor.java:52) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.processor.RedeliveryErrorHandler.doStart(RedeliveryErrorHandler.java:1472) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ChildServiceSupport.start(ChildServiceSupport.java:44) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ChildServiceSupport.start(ChildServiceSupport.java:31) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.processor.interceptor.DefaultChannel.doStart(DefaultChannel.java:160) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:62) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.processor.MulticastProcessor.doStart(MulticastProcessor.java:1186) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.processor.WrapProcessor.doStart(WrapProcessor.java:52) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.processor.RedeliveryErrorHandler.doStart(RedeliveryErrorHandler.java:1472) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ChildServiceSupport.start(ChildServiceSupport.java:44) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ChildServiceSupport.start(ChildServiceSupport.java:31) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.processor.interceptor.DefaultChannel.doStart(DefaultChannel.java:160) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:62) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.processor.MulticastProcessor.doStart(MulticastProcessor.java:1186) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:60) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:104) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startServices(ServiceHelper.java:90) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.processor.DelegateAsyncProcessor.doStart(DelegateAsyncProcessor.java:80) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:75) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.RouteService.startChildService(RouteService.java:370) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.RouteService.doWarmUp(RouteService.java:196) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.RouteService.warmUp(RouteService.java:145) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	... 36 common frames omitted
Caused by: org.apache.camel.NoSuchBeanException: No bean could be found in the registry for: connector-rest-swagger-http4 of type: org.apache.camel.spi.RestProducerFactory
	at org.apache.camel.component.rest.RestEndpoint.createProducer(RestEndpoint.java:316) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.component.rest.swagger.RestSwaggerEndpoint.createProducerFor(RestSwaggerEndpoint.java:286) ~[camel-rest-swagger-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.component.rest.swagger.RestSwaggerEndpoint.createProducer(RestSwaggerEndpoint.java:191) ~[camel-rest-swagger-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.component.connector.DefaultConnectorEndpoint.createProducer(DefaultConnectorEndpoint.java:51) ~[camel-connector-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	at org.apache.camel.impl.ProducerCache.doGetProducer(ProducerCache.java:573) ~[camel-core-2.21.0.fuse-720029.jar!/:2.21.0.fuse-720029]
	... 99 common frames omitted
2018-11-12 11:38:05.704  INFO 1 --- [           main] ationConfigEmbeddedWebApplicationContext : Closing org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext@40e6dfe1: startup date [Mon Nov 12 11:37:55 UTC 2018]; root of context hierarchy
2018-11-12 11:38:05.706  INFO 1 --- [           main] o.s.c.support.DefaultLifecycleProcessor  : Stopping beans in phase 2147483647
2018-11-12 11:38:05.706  INFO 1 --- [           main] o.s.c.support.DefaultLifecycleProcessor  : Stopping beans in phase 0
2018-11-12 11:38:05.707  INFO 1 --- [           main] o.s.j.e.a.AnnotationMBeanExporter        : Unregistering JMX-exposed beans on shutdown
```
[i-amq-to-todo-integration-3-nhmzs.log](https://github.com/syndesisio/syndesis/files/2571768/i-amq-to-todo-integration-3-nhmzs.log)

--- a/app/connector/rest-swagger/src/main/resources/META-INF/spring.factories
+++ b/app/connector/rest-swagger/src/main/resources/META-INF/spring.factories
@@ -1,3 +1,3 @@
-
 org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
-io.syndesis.connector.rest.swagger.springboot.SwaggerConnectorConnectorAutoConfiguration
+  io.syndesis.connector.rest.swagger.springboot.SwaggerConnectorConnectorAutoConfiguration,\
+  io.syndesis.connector.rest.swagger.springboot.SwaggerConnectorConfiguration

